Test/CaptureFileWriterTest.cs

```diff
@@ -17,12 +21,57 @@ public CaptureFileWriterTest()
         [Test]
         public void TestFileCreationAndDeletion()
         {
-            using (var wd = new CaptureFileWriterDevice(@"abc.pcap"))
+            using (var wd = new CaptureFileWriterDevice(filename))
             {
-                wd.Write(new byte[] { 1, 2, 3, 4 });
+                Assert.AreEqual(filename, wd.Name);
+                Assert.IsNotEmpty(wd.Description);
+                var bytes = new byte[] { 1, 2, 3, 4 };
+                wd.Write(bytes);
+
+                var p = new RawCapture(PacketDotNet.LinkLayers.Ethernet, new PosixTimeval(), bytes);
+                wd.Write(p);
             }
             System.IO.File.Delete(@"abc.pcap");
         }
+
+        [Test]
+        public void TestCreationOptions()
+        {
+            // valid arguments results in the object being created
+            using var valid = new CaptureFileWriterDevice(PacketDotNet.LinkLayers.Ethernet, null, "somefilename.pcap", System.IO.FileMode.Open);
+
+            // invalid snapshot length should throw
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                using var wd = new CaptureFileWriterDevice(PacketDotNet.LinkLayers.Ethernet, 500000, "somefilename.pcap", System.IO.FileMode.Open);
+            });
+
+            // file mode of append should throw
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                using var wd = new CaptureFileWriterDevice(PacketDotNet.LinkLayers.Ethernet, null, "somefilename.pcap", System.IO.FileMode.Append);
+            });
+        }
+
+        [Test]
+        public void SendPacket()
+        {
+            var bytes = new byte[] { 0x10, 0x20, 0x30 };
+
+            using (var wd = new CaptureFileWriterDevice(filename))
+            {
+                Assert.Throws<NotSupportedOnCaptureFileException>(() => wd.SendPacket(bytes));
+            }
+        }
+
+        [Test]
+        public void Statistics()
+        {
+            using (var wd = new CaptureFileWriterDevice(filename))
+            {
+                Assert.Throws<NotSupportedOnCaptureFileException>(() => { var s = wd.Statistics; });
```

Comment:
> AVOID throwing exceptions from property getters.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property

Reply:
What about modifying CaptureFileReader and writer devices to return an empty statistics object instead? The idea behind the exception was that you can't get any statistics from the class. I figured it was a more robust approach to throw so the user wouldn't be confused why they weren't getting any stats. Would seem a shame to make it a method just because it throws in these cases. For supported devices it doesn't throw in normal operation.

Reply:
You could simply return null

Reply:
That's a good idea. Let me make that change in another commit on that same branch.